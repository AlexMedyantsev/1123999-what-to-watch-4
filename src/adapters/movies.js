const getLevel = (score) => {
  let level = ``;

  if (score < 3) {
    level = `Bad`;
  } else if (score >= 3 && score < 5) {
    level = `Normal`;
  } else if (score >= 5 && score < 8) {
    level = `Good`;
  } else if (score >= 8 && score < 10) {
    level = `Very Good`;
  } else if (score === 10) {
    level = `Awesome`;
  }

  return level;
};

export const adaptMovies = (moviesFromServer) => {
  return moviesFromServer.map((movie) => {
    return ({
      image: movie.preview_image,
      posterSrc: movie.poster_image,
      bgSrc: movie.background_image,
      title: movie.name,
      genre: movie.genre,
      year: movie.released,
      score: movie.rating,
      level: getLevel(movie.rating),
      movieLink: movie.video_link,
      previewLink: movie.preview_video_link,
      scoresCount: movie.scores_count,
      description: movie.description,
      director: movie.director,
      starring: movie.starring,
      runTime: movie.run_time,
      link: `movie-page.html`,
      id: movie.id,
      backgroundColor: movie.background_color,
      isFavorite: movie.is_favorite
    });
  });
}

export const adaptMovie = (movie) => {
  return ({
    image: movie.preview_image,
    posterSrc: movie.poster_image,
    bgSrc: movie.background_image,
    title: movie.name,
    genre: movie.genre,
    year: movie.released,
    score: movie.rating,
    level: getLevel(movie.rating),
    movieLink: movie.video_link,
    previewLink: movie.preview_video_link,
    scoresCount: movie.scores_count,
    description: movie.description,
    director: movie.director,
    starring: movie.starring,
    runTime: movie.run_time,
    link: `movie-page.html`,
    id: movie.id,
    backgroundColor: movie.background_color,
    isFavorite: movie.is_favorite
  });
};
