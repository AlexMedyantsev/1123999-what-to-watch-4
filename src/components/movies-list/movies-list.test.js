import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list.jsx";
import movies from "../../mocks/movies.js";

describe(`MovieList snapshot`, () => {
  it(`Should movies list render correctly`, () => {
    const tree = renderer
      .create(<MoviesList
        movies={movies}
        onMovieCardClick={() => {}}
      />, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
