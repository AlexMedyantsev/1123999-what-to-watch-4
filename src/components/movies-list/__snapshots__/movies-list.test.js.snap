// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList snapshot Should movies list render correctly 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
    >
      <video>
        Sorry, this preview is not available in your browser
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Name
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
    >
      <video>
        Sorry, this preview is not available in your browser
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Name
      </a>
    </h3>
  </article>,
]
`;
